tests/src/test/java/org/eclipse/hono/tests/coap/CoapTestBase.java

```diff
@@ -1311,6 +1318,141 @@ public void testUploadMessagesWithTtdThatReplyWithOneWayCommand(
                 });
     }
 
+    /**
+     * Ensures that a timestamp is within 2 seconds of current system clock.
+     * This is necessary because the test containers may have a clock drift of a few milliseconds
+     * from the host clock.
+     *
+     * @param ts the timestamp to check
+     */
+    private void assertTimestampWithinRangeOfCurrentTime(final long ts) {
+        // Use a buffer time of 2 seconds
+        final long buffer = 2000;
+        final long currentTime = System.currentTimeMillis();
+        assertThat(ts).isIn(Range.closed(currentTime - buffer, currentTime + buffer));
+    }
+
+    /**
+     * Verify that the CoAP adapter responds with a time option in the response when request includes
+     * a time option.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    @Timeout(value = 10, timeUnit = TimeUnit.SECONDS)
+    public void testServerTimeResponseWithRequestOption(final VertxTestContext ctx) {
+        final Checkpoint checks = ctx.checkpoint(2);
+        helper.registry.addPskDeviceForTenant(tenantId, new Tenant(), deviceId, SECRET)
+                .compose(response -> createConsumer(tenantId, msg -> {
+                    logger.trace("received {}", msg);
+                    ctx.verify(() -> {
+                        DownstreamMessageAssertions.assertTelemetryApiProperties(msg);
+                        DownstreamMessageAssertions.assertMessageContainsAdapterAndAddress(msg);
+                        assertThat(msg.getPayload().length()).isGreaterThan(0);
+                        assertAdditionalMessageProperties(msg);
+                    });
+                    checks.flag();
+                }))
+                .compose(ok -> {
+                    final CoapClient client = getCoapsClient(deviceId, tenantId, SECRET);
+                    final Promise<CoapResponse> result = Promise.promise();
+                    final var request = createCoapsRequest(Code.POST, getPostResource(), 0);
+                    request.getOptions().addOtherOption(new Option(TIME_OPTION_NUMBER, new byte[0]));
+                    client.advanced(getHandler(result, ResponseCode.CHANGED), request);
+                    return result.future();
+                })
+                .onSuccess(res -> {
+                    ctx.verify(() -> {
+                        assertThat(res.getOptions().hasOption(TIME_OPTION_NUMBER)).isTrue();
+                        final long serverTimeValue = res.getOptions().getOtherOption(TIME_OPTION_NUMBER).getLongValue();
+                        assertTimestampWithinRangeOfCurrentTime(serverTimeValue);
+                    });
+                    checks.flag();
+                })
+                .onFailure(ctx::failNow);
+    }
+
+    /**
+     * Verify that the CoAP adapter responds with a time option in the response when request includes
+     * the "hono-time" query parameter.
```

Comment:
maybe refer to the constant definition instead?

Reply:
Absolutely... when I figure out the correct way to add the dependency to the coap adapter module.

Reply:
figured out